src/ActivityRepository.js

```diff
@@ -1,21 +1,25 @@
-class ActivityRepository {
-  constructor(id, activityData) {
+import UtilityRepository from "../src/UtilityRepository";
+
+class ActivityRepository extends UtilityRepository {
+  constructor(id, activityData, sleepData, hydrationData, userData) {
+    super(userData, sleepData, hydrationData, activityData)
     this.id = id;
     this.activityData = activityData;
-    this.user = this.getUserLogs(id);
+    this.user = this.getUserLogs(id, "activityData")
   }
 
-  getUserLogs(id) {
-    return this.activityData.filter(user => user.userID === id);
-  }
+  // getUserLogs(id) {
+    // return this.activityData.filter(user => user.userID === id);
+  //   this.getUserLogs(id, this.activityData)
+  // }
 
   getUserDate(date) {
```

Comment:
I think this might be another chance to use the getUserInfoByDate super method from the parent. @VeeAndrade ,

Reply:
Sounds good, I'm taking it out and will implement it in the index.js.